telegram-bot-api/Client.cpp

```diff
@@ -156,11 +171,31 @@ void Client::fail_query_with_error(PromisedQueryPtr &&query, object_ptr<td_api::
   fail_query_with_error(std::move(query), error->code_, error->message_, default_message);
 }
 
-Client::Client(td::ActorShared<> parent, const td::string &bot_token, bool is_test_dc, int64 tqueue_id,
+Client::Client(td::ActorShared<> parent, const td::string &bot_token, bool is_user, bool is_test_dc, int64 tqueue_id,
                std::shared_ptr<const ClientParameters> parameters, td::ActorId<BotStatActor> stat_actor)
     : parent_(std::move(parent))
     , bot_token_(bot_token)
     , bot_token_id_("<unknown>")
+    , is_user_(is_user)
+    , is_test_dc_(is_test_dc)
+    , tqueue_id_(tqueue_id)
+    , parameters_(std::move(parameters))
+    , stat_actor_(std::move(stat_actor)) {
+  messages_lru_root_.lru_next = &messages_lru_root_;
+  messages_lru_root_.lru_prev = &messages_lru_root_;
+
+  static auto is_inited = init_methods();
+  CHECK(is_inited);
+}
+
+Client::Client(td::ActorShared<> parent, const td::string &bot_token, const td::string &phone_number, bool is_test_dc,
+               int64 tqueue_id, std::shared_ptr<const ClientParameters> parameters,
+               td::ActorId<BotStatActor> stat_actor)
+    : parent_(std::move(parent))
+    , bot_token_(bot_token)
+    , bot_token_id_("<unknown>")
+    , phone_number_(phone_number)
+    , is_user_(true)
```

Comment:
Shouldn't this default to false?
Since the user mode is a new addition we should default to the previous mode (that is assume is a bot)

Reply:
I have to constructors for Client. This constructor is only called in the userlogin method and passes a phone number, so it is automatically a user. The other constructor is used when logging in as a bot.

Reply:
i'd prefer to say explicitally if it is an user and assume that is a bot otherwise to preserve the original functionalities (as it may break the upstream)
awaiting for @andrew-ld to review this

Reply:
The other Constructor explicitely says if it is an user and this Constructor can not be called for bots because of the phone number argument. But I can create a parameter that always passes true for is_user.

Reply:
@code1mountain change `is_user_(true)` to `is_user_(false)` and call the explicit costructor when the user mode is needed, this way we can preserve backward compatibility

Reply:
I think there is a misunderstanding: The two Constructors are for different tasks.

1. The first Constructor is for waking up already authorized Clients or creating new Bot Clients. This is the default Constructor that has a new argument `is_user`

2. (This one) The second Constructor is only used when creating a new User Client. Therefore it can be automatically assumed that the Client is a User.